### DISCUSSION TITLE
[Feature Request] Adjustable column widths in advanced web interface

### DISCUSSION THREAD

**Discussion by andrewanderson** [mastodon/mastodon#21453] [General]
The advanced web interface has a lot of potential, but on a widescreen it makes very poor use of the horizontal space unless you add many columns.  Column width should either:

- Be manually adjustable
- Automatically adjust based on the number of columns you have

**Comment by VK2AAF** [comment]
Great idea. The Better TweetDeck extension for TweetDeck could be a model for this feature request. Column widths are specified in pixels. I'm aware of several Mastodon users who would use this feature. 
![image](https://user-images.githubusercontent.com/119258058/204157822-183d9707-a539-4f62-b9d3-ba46800e6a1e.png)

**Comment by staringatclouds** [comment]
As an interim  measure I'd like to suggest making the "drawer" class width 19% & the "column" class width 27% instead of the fixed number of pixels they are now

NB these percentage values work for me but may not suit everyone

That way the columns will expand & contract with the display & will fill the available space

Ideally I'd like to resize columns individually as andrewanderson suggested in the first post but that's more work & resources are limited, this is a cheap & cheerful interim measure

I use the browser debugging tools (right click on element & select "Inspect" in Firefox) to set these values manually & they don't appear to break anything

This of course doesn't mean I haven't broken something by doing it, but nothing is leaping off the page

However I need to do this every time I load the page as they're not persistent values

**Reply by AsiaHand** [reply]
The Resize-O-Don user script provides adjustable column width on Firefox and was a gamechanger for me. I agree it would be good to have this as standard! Details here: https://h-net.social/@arose62@aus.social/109559909637422193 and installation instructions in the replies.

**Comment by grahamperrin** [comment]
> Adjustable column widths in advanced web interface

Please subscribe to this issue: 

* https://github.com/mastodon/mastodon/issues/8632
